Add `grid_spacing` and `origin_coords` to the fields of `ImagingPlane`

h-mayorquin opened the pull request:
## Motivation

This should fix #1885 

I discussed this with @rly and it seems that it was just an omission.  This PR tries to address this. So far, this seems to work for `grid_spacing` but for some reason that escapes me it seems that `origin_coords` is not added to the fields. 

![image](https://github.com/NeurodataWithoutBorders/pynwb/assets/6429509/5ea3ea40-3cc2-43d0-a49e-c119ae4b9e6f)


```python
from pynwb.testing.mock.ophys import mock_ImagingPlane

imaging_plane = mock_ImagingPlane(grid_spacing=[1.0, 1.0], origin_coords=[0.0, 0.0])
imaging_plane.origin_coords
list(imaging_plane.fields.keys())

```


## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

h-mayorquin committed:
add grid spacing and origin_coords to fields

rly committed:
Update ophys.py

rly committed:
Merge branch 'dev' into add_grid_spacing_to_fields

rly commented:
> So far, this seems to work for grid_spacing but for some reason that escapes me it seems that origin_coords is not added to the fields.

@h-mayorquin I think you were missing a comma in the `__nwbfields__` list. I fixed that and I think it works now. Could you check, and if it looks good, add a changlog?

h-mayorquin committed:
add changelog

h-mayorquin commented:
> > So far, this seems to work for grid_spacing but for some reason that escapes me it seems that origin_coords is not added to the fields.
> 
> @h-mayorquin I think you were missing a comma in the `__nwbfields__` list. I fixed that and I think it works now. Could you check, and if it looks good, add a changlog?

My bad, I tested your change and it is working now. Thanks.

Added the changelog.

rly committed:
Merge branch 'dev' into add_grid_spacing_to_fields

rly committed:
Update CHANGELOG.md

rly commented:
Thanks for the fix @h-mayorquin !

rly committed:
Merge branch 'dev' into add_grid_spacing_to_fields
